### PR TITLE
Zxc burner contract.

### PR DESCRIPTION
**Pull request by MoMannn**

### CODE REVIEW

**Comment by xpepermint** [comment]
# [Codecov](https://codecov.io/gh/0xcert/framework/pull/666?src=pr&el=h1) Report
> Merging [#666](https://codecov.io/gh/0xcert/framework/pull/666?src=pr&el=desc) into [master](https://codecov.io/gh/0xcert/framework/commit/66a3a30d6ceb4ba7f87806eb2f4329089bd3f989&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/0xcert/framework/pull/666?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #666   +/-   ##
=======================================
  Coverage   79.49%   79.49%           
=======================================
  Files         129      129           
  Lines        2839     2839           
  Branches      334      334           
=======================================
  Hits         2257     2257           
  Misses        420      420           
  Partials      162      162           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/0xcert/framework/pull/666?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/0xcert/framework/pull/666?src=pr&el=footer). Last update [66a3a30...9855cec](https://codecov.io/gh/0xcert/framework/pull/666?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by MoMannn** [comment]
> This repeats/inlines code from other contracts. One class per file please.

This is directly copied code from deployed mainnet ZXC code and modified so it works with 0.6.1 solidity. This contract will directly communicate with existing ZXC code to allow anyone to burn tokens and remove any contract ownership. This is the reason for one file code.

**Comment by fulldecent** [comment]
Got it. So the existing contract is used just as an interface to work with deployed contract. (And also so you can load the contract into your test environment.)

That's not a contract mock, that's a test mock. It should be in the other folder.

Can add to top of test mock:

> This is the exact source code for the deployed ZXC contract, deployed on Ethereum Mainnet at 0x83e2be8d114f9661221384b3a50d24b96a5653f5 on 2018-05-??.
> Brief modifications are made just so this can compile with the latest Solidity.
> But otherwise it is kept as similar as possible for test cases here.

And in the contracts folder can add ZXC interface Solidity file.

And then the zxc-burner.sol can reference the ZXC interface rather than the ZXC contract implementation.

Otherwise, everything looks good to go.

**Comment by xpepermint** [comment]
@fulldecent you are right. It's a test mock. @MoMannn I agree with William's proposal. Let's refactor this as explained.

**Comment by xpepermint** [comment]
Btw, guys ... do you believe in evil spirits? We are `burning` tokens under PR number `666` :).

File: packages/0xcert-ethereum-erc20-contracts/src/contracts/zxc-burner.sol [hunk 1]
```diff
@@ -0,0 +1,48 @@
+pragma solidity 0.6.1;
+
+import "./zxc.sol";
+
+/**
+ * @title This is a contract that allows anyone to burn ZXC tokens.
+ * @dev This contract has ownership over ZXC smart contract and as such allows anyone to burn ther
```
Comment (fulldecent) [inline packages/0xcert-ethereum-erc20-contracts/src/contracts/zxc-burner.sol]: their

**Comment by fulldecent** [comment]
😈

**Pull request merged**